pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/LLCSegmentCommitUpload.java

```diff
@@ -111,16 +121,14 @@ boolean uploadSegment(final String instanceId, final String segmentNameStr, fina
 
       final String rawTableName = segmentName.getTableName();
       final File tableDir = new File(baseDataDir, rawTableName);
-      String generatedSegmentFileName = SegmentCompletionUtils.generateSegmentFileName(segmentNameStr, offset, instanceId);
-      final File segmentFile = new File(tableDir, generatedSegmentFileName);
-
-      // Always delete the segment file if it exists for split commit.
+      // For split commit, generate a new segment name with offset, instanceId, and UUID
+      _generatedSegmentName = SegmentCompletionUtils.generateSegmentFileName(segmentNameStr);
+      final File segmentFile = new File(tableDir, _generatedSegmentName);
       if (segmentFile.exists()) {
-        LOGGER.warn("Segment file {} exists. Replacing with upload from {}", segmentNameStr, instanceId);
-        FileUtils.deleteQuietly(segmentFile);
+        LOGGER.warn("Segment file exists " + segmentFile);
```

Comment:
checking is ok, but if uuid is correct, we should never have the segment file here. add a comment to that effect